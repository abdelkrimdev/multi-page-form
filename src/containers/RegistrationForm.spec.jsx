import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import ConnectedRegistrationForm, { RegistrationForm } from './RegistrationForm'

const mockStore = configureStore()

describe('registration form', () => {
  it('should render without crashing.', () => {
    const wrapper = shallow(
      <RegistrationForm currentStep={0} />
    )

    expect(wrapper).toBeDefined()
  })

  it('should render correctly.', () => {
    const wrapper = shallow(<RegistrationForm currentStep={0} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should connect to the store without crashing.', () => {
    const store = mockStore({
      form: { step: 0 }
    })

    const container = shallow(<ConnectedRegistrationForm store={store} />).dive()

    expect(container).toBeDefined()
  })
})
