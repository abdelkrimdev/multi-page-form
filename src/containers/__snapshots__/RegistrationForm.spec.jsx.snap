// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registration form should render correctly. 1`] = `
<Fragment>
  <Steps
    current={0}
    iconPrefix="ant"
    prefixCls="ant-steps"
  >
    <Step
      key="1"
      title="Full Name"
    />
    <Step
      key="2"
      title="Email Address"
    />
    <Step
      key="3"
      title="Phone Number"
    />
    <Step
      key="4"
      title="Salary Indication"
    />
  </Steps>
  <div
    className="steps-content"
  >
    <FullName
      onChange={[Function]}
    />
  </div>
  <div
    className="steps-action"
  >
    <Button
      block={false}
      ghost={false}
      loading={false}
      onClick={[Function]}
      prefixCls="ant-btn"
      type="primary"
    >
      Next
    </Button>
  </div>
</Fragment>
`;
